PyInstaller/building/imphook.py

```diff
@@ -12,18 +12,449 @@
 Code related to processing of import hooks.
 """
 
-import glob
+import glob, weakref
 import os.path
 
 from .. import log as logging
-from .utils import format_binaries_and_datas
-from ..compat import expand_path
-from ..compat import importlib_load_source
+from ..compat import expand_path, importlib_load_source, FileNotFoundError
 from .imphookapi import PostGraphAPI
+from .utils import format_binaries_and_datas
 
 logger = logging.getLogger(__name__)
 
 
+class ModuleHookCache(dict):
```

Comment:
> Is dict.copy() really called frequently enough to warrant this sort of aggressive optimization?

Probably not, but that's just one part of it. The real winners are probably list catenation, set intersection, difference, dict update and the like.